### PR TITLE
docs(man): write about guest parsers

### LINKED ISSUE #3007: Write about kind collisions to ctags-faq(7)
> @masatake - it seems that the c tag was already used for classes in Python. Can you update the merge to use capitalized C for code cells instead?

See https://github.com/universal-ctags/ctags/issues/2978#issuecomment-829451352 .
See also https://docs.ctags.io/en/latest/man/ctags-client-tools.7.html#multiple-languages-for-an-input-file.

Use the `language` field to distinguish whether a tag having c as its kind means `class` in the Python or` cell` in the IPythonCell.
`--fields=+l` or `--fields=+{language}'.

I will not update. Even if I will update as you wrote, your issue will occur again when `C` kind is introduced to the Python parser.

I recommend you to enable `--fields=+Kzl`.

The rest of the work is up to the developers of your client tool.

I must add this to ctags-faq(7).

_Originally posted by @masatake in https://github.com/universal-ctags/ctags/issues/3000#issuecomment-833126017_

### LINKED ISSUE #3888: docs(man): write about using multiple parsers for single input file
> I will try to document how to support a tag file of "Running multiple parsers on an input file" in a client tool by extending the man page. 

Cool. Thanks for your attention.

_Originally posted by @andregpss in https://github.com/universal-ctags/ctags/issues/3802#issuecomment-1733820436_

### PR DESCRIPTION
**Pull request by masatake**
Close #3888
Close #3007

### CODE REVIEW

**Review by hirooih — CHANGES_REQUESTED** [review]
It is very clearly written in general.
I commented on the questions I had when I read it.

A typo should be fixed.

Others are comments.

File: docs/man/ctags-client-tools.7.rst [hunk 1]
```diff
@@ -325,30 +325,130 @@ for using notable ones.
...
+	!_TAG_KIND_DESCRIPTION!CSS	c,class	/classes/
+	...
+	!_TAG_KIND_DESCRIPTION!HTML	c,class	/classes/
+	!_TAG_KIND_DESCRIPTION!HTML	h,heading1	/H1 headings/
+	...
+	!_TAG_KIND_DESCRIPTION!JavaScript	c,class	/classes/
+	...
+
+From the seoncd field of the output, a tool can know the mapping
+between a kind single-letter and a kind long-name.
```
[the remark this answers is not in the document]
Comment (masatake) [reply docs/man/ctags-client-tools.7.rst:430]: How about "a single-letter for a kind and a long-name for the kind" ?

File: docs/man/ctags-client-tools.7.rst [hunk 2]
```diff
@@ -325,30 +325,130 @@ for using notable ones.
...
+	...
+	!_TAG_KIND_DESCRIPTION!CSS	c,class	/classes/
+	...
+	!_TAG_KIND_DESCRIPTION!HTML	c,class	/classes/
+	!_TAG_KIND_DESCRIPTION!HTML	h,heading1	/H1 headings/
+	...
+	!_TAG_KIND_DESCRIPTION!JavaScript	c,class	/classes/
+	...
+
+From the seoncd field of the output, a tool can know the mapping
```
[the remark this answers is not in the document]
Comment (masatake) [reply docs/man/ctags-client-tools.7.rst:429]: Thank you.

File: docs/man/ctags-client-tools.7.rst [hunk 3]
```diff
@@ -325,30 +325,130 @@ for using notable ones.
...
-	title	/tmp/foo.html	/^  <h1>title<\/h1>$/;"	h	language:HTML
-	x	/tmp/foo.html	/var x = 1/;"	v	language:JavaScript
+	$ ctags -o - --extras='{guest}' --fields=+'{language}' input.html
+	MyObject	input.html	/class MyObject {}/;"	c	language:JavaScript
+	h1.heading	input.html	/h1.heading { color: red; }/;"	c	language:CSS
+	title	input.html	/^<h1 class='heading'>title<\/h1>$/;"	h	language:HTML
+
+If the client tool knows what is ``c`` in JavaScript and what is ``c`` in CSS,
+the ``language:`` field provides enough information. If not, enabling ``K`` field with
+``--fields=+K`` may be helpful.
```
[the remark this answers is not in the document]
Comment (masatake) [reply docs/man/ctags-client-tools.7.rst:391]: Thank you.

Surprisingly (and sadly), many client tools hardcode single-letter kinds and long-name kinds. Such a tool can say what is ``c`` in JavaScript input. The tool also knows the language for `*.js` file. I'll revise the sentences you pointed out.

File: docs/man/ctags-client-tools.7.rst [hunk 4]
```diff
@@ -325,30 +325,130 @@ for using notable ones.
...
+
+	$ ctags -o - --extras='{guest}' --fields=+'{language}K' input.html
+	MyObject	/tmp/input.html	/class MyObject {}/;"	class	language:JavaScript
+	h1.heading	/tmp/input.html	/h1.heading { color: red; }/;"	class	language:CSS
+	title	/tmp/input.html	/^<h1 class='heading'>title<\/h1>$/;"	heading1	language:HTML
+
+The kind fields are represented in long-names instead of single-letters.
+
+Enabling ``z`` field with ``--fields=+z`` makes the tag file larger but it helps a client tool
+parse tags output:
```
[the remark this answers is not in the document]
Comment (masatake) [reply docs/man/ctags-client-tools.7.rst:403]: I will remove the description about ``--fields=+z``.

File: docs/man/ctags-client-tools.7.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (hirooih) [reply docs/man/ctags-client-tools.7.rst]: It is better than mine.

**Review by hirooih — APPROVED** [review]

**Comment by hirooih** [comment]
I could read very smoothly.
Thanks.

**Comment by masatake** [comment]
@hirooih Thank you.

**Pull request merged**